drivers/input/touchscreen/Kconfig

```diff
@@ -977,6 +977,15 @@ config TOUCHSCREEN_SYNAPTICS_I2C_RMI4
 
 	  To compile this driver as a module, choose M here: the
 	  module will be called synaptics_i2c_rmi4.
+	  
+config TOUCHSCREEN_SYNAPTICS_UNOFFICIAL
+	tristate "Unofficial Synaptics DSX I2C touchscreen"
```

Comment:
You never check the value of this config option, only wether it's defined or not, so `tristate` is probably not the correct type for it

Reply:
I will take a look later.